kubelet: fix setup of emptyDir with sizeLimit (fsquota)

alex-matei opened the pull request:
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When size limit is specified subsequent invocations will fail because ibytes is changed to -1 and stored internally in quotaSizeMap during the first call.  Later invocation will see that the requested size doesn't match the actual stored value and it will fail.

#### Which issue(s) this PR fixes:
Fixes #117098

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs

```

alex-matei committed:
kubelet: fix setup of emptyDir with sizeLimit (fsquota)

When size limit is specified subsequent invocations will fail because
ibytes is changed to -1 and stored internally in quotaSizeMap during the
first call.  Later invocation will see that the requested size doesn't
match the actual stored value and it will fail.

Signed-off-by: Alexandru Matei <alexandru.matei@uipath.com>

alex-matei commented:
/cc @pacoxu

msau42 commented:
/ok-to-test

pacoxu commented:
The UT needs to be fixed and can we address the problem with a unit test or e2e test? Then we can avoid getting a regression for this issue.

alex-matei commented:
@pacoxu Done, the test that was failing was exactly this case, calling AssignQuota with exactly the same size again, fixed it. I added a new test case for calling AssignQuota again with a different size, this one should fail all the time.

pacoxu commented on pkg/volume/util/fsquota/quota_linux_test.go:
```diff
@@ -461,6 +461,10 @@ var quotaTestCases = []quotaTestCase{
 	},
 	{
```
At a first glance of the failed ut, I did not realize that this test case is about setting a limitation smaller than the current quota.

Should we add a testName for the quotaTestCase struct to make it clear?

pacoxu commented:
Logically, this LGTM. 
I need do some further test on it.

pacoxu commented:
/lgtm
/cc @jsafrane

pacoxu commented:
/priority important-soon
/triage accepted

jsafrane commented:
/approve
